distribution/src/bin/opensearch-env

```diff
@@ -40,9 +49,12 @@ if [ ! -z "$JAVA_HOME" ]; then
   JAVA="$JAVA_HOME/bin/java"
   JAVA_TYPE="JAVA_HOME"
 else
-  if [ "$(uname -s)" = "Darwin" ]; then
-    # macOS has a different structure
+  if [ $OS = "darwin" ]; then
+    # macOS bundled Java
     JAVA="$OPENSEARCH_HOME/jdk.app/Contents/Home/bin/java"
+  elif [ $OS = "freebsd" ]; then
```

Comment:
Why wouldn't we be using the packaged Java version on FreeBSD?

Reply:
Same as for Node, Linux ELF isn't compatible and JAVA_HOME afaik not set by default on FreeBSD

Reply:
I see how this makes things better, but we do bundle a JDK under `$OPENSEARCH_HOME`, shouldn't we be bundling a compatible JDK, and releasing a FreeBSD package instead of doing this? I'm thinking out loud, so bear with my FreeBSD ignorance.

Reply:
The best as far as FreeBSD is concerned is probably to ship a tarball which only contains the built java objects, and does not bundle the JDK (so a platform agnostic tarball).

It is not uncommon for FreeBSD ports to download such tarballs to build packages by just adding dependencies to the relevant ports (e.g. openjdk14, node), extracting this archive at the right location and providing utility scripts (e.g. rc.d init file).